e2e/wdio/headless/context-management.e2e.ts

```diff
@@ -0,0 +1,67 @@
+/// <reference types="@wdio/lighthouse-service" />
+import { browser, expect } from '@wdio/globals'
+
+describe('context management', () => {
+    it('should switch between contexts when the window changes', async () => {
+        await browser.url('http://guinea-pig.webdriver.io/')
+
+        await browser.newWindow('https://webdriver.io')
+        await expect($('.hero__subtitle')).toBePresent()
+        await expect($('.red')).not.toBePresent()
+
+        await browser.switchWindow('guinea-pig.webdriver.io')
+        await expect($('.red')).toBePresent()
+        await expect($('.hero__subtitle')).not.toBePresent()
+
+        await browser.switchWindow('Next-gen browser and mobile automation test framework for Node.js')
+        await expect($('.hero__subtitle')).toBePresent()
+        await expect($('.red')).not.toBePresent()
+    })
+
+    async function switchToFrameBySelector(selector: string) {
+        const iframe = await browser.findElement('css selector', selector)
+        await browser.switchToFrame(iframe)
+    }
+    async function setAttribute(selector: string, attributeName: string, attributeValue: string) {
+        await browser.execute((elementSelector, attributeName, attributeValue) => {
```

Comment:
Would be easier to used elem.execute here so you already have the element reference as in v9 the execute method also exists on the Element object